src/rgw/rgw_rest_sts.cc

```diff
@@ -201,6 +201,7 @@ RGWOp *RGWHandler_REST_STS::op_post()
            s->info.args.append(key, value);
          }
        }
+       s->info.args.append("data", post_body);
```

Comment:
would it make sense to calculate the hash here and store that, instead of copying the body to hash later?

Reply:
I'll check if that is possible.

Reply:
@cbodley : Have made the change.

Reply:
looks good!